pkg-manager/resolve-dependencies/src/index.ts

```diff
@@ -304,6 +307,10 @@ export async function resolveDependencies (
   }
 
   newLockfile.catalogs = getCatalogSnapshots(Object.values(resolvedImporters).flatMap(({ directDependencies }) => directDependencies))
+  for (const alias in newDefaultCatalogs) {
+    newLockfile.catalogs.default ??= {}
+    newLockfile.catalogs.default[alias] = newDefaultCatalogs[alias]
+  }
```

Comment:
At first glance, it's a bit surprising that there's a new block modifying `newLockfile.catalogs` after `getCatalogSnapshots` runs. I agree it's necessary.

To make this clearer to future readers, could you add a comment that this is to support new catalog entries that are added as part of the installation, and this requires special handling since these dependencies weren't originally resolved with the `catalog:` protocol? Thanks.

Reply:
https://github.com/pnpm/pnpm/pull/9484/commits/918f6b5e05de4a693b35df3431246c0fedca37a6